### PR TITLE
Website: Release 15.0.1 blog post

### PR DESCRIPTION
**Pull request by raulcd**
Release blog post for 15.0.1

### CODE REVIEW

File: _posts/2024-03-07-15.0.1-release.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## C++ notes
+
+## Java notes
+
+## Python notes
```
Comment (raulcd) [inline _posts/2024-03-07-15.0.1-release.md:41]: @jorisvandenbossche @AlenkaF could you add details for the included fixes?

File: _posts/2024-03-07-15.0.1-release.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+from [**18 distinct contributors**][2]. See the Install Page to learn how to
+get the libraries for your platform.
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## C++ notes
+
+## Java notes
```
Comment (raulcd) [inline _posts/2024-03-07-15.0.1-release.md:39]: @danepitkin can you add the Java issue that triggered the new release?

File: _posts/2024-03-07-15.0.1-release.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+The Apache Arrow team is pleased to announce the 15.0.1 release.
+This is mostly a bugfix release that includes [**42 resolved issues**][1]
+from [**18 distinct contributors**][2]. See the Install Page to learn how to
+get the libraries for your platform.
+
+The release notes below are not exhaustive and only expose selected highlights
+of the release. Other bugfixes and improvements have been made: we refer
+you to the [complete changelog][3].
+
+## C++ notes
```
Comment (raulcd) [inline _posts/2024-03-07-15.0.1-release.md:37]: @pitrou any not worth adding to the blog post?

**Review by raulcd — COMMENTED** [review]
This is the full changelog if it helps you to fill the notes: https://arrow.apache.org/release/15.0.1.html#changelog

File: _posts/2024-03-07-15.0.1-release.md (hunk 1 above)
Comment (AlenkaF) [reply _posts/2024-03-07-15.0.1-release.md:41]:
```suggestion
## Python notes

* Fix race condition with concurrent invocation of `_pandas_api.is_data_frame(df)` ([GH-39313](https://github.com/apache/arrow/issues/39313))
* Fix leaking references to Numpy dtypes ([GH-39599](https://github.com/apache/arrow/issues/39599))
* Fix except clauses in order to be compatible with Cython 3.0.9 ([GH-40386](https://github.com/apache/arrow/issues/40386))
* Fix interpreter deadlock when using `GeneratorStream` ([GH-40004](https://github.com/apache/arrow/issues/40004))

```

**Review by raulcd — COMMENTED** [review]
:+1:

**Pull request merged**